softlayer/vm/softlayer_creator.go

```diff
@@ -93,3 +134,40 @@ func (c SoftLayerCreator) resolveNetworkIP(networks Networks) (string, error) {
 
 	return network.IP, nil
 }
+
+func (c SoftLayerCreator) parseMbusURL(mbusURL string, primaryBackendIpAddress string) (string, error) {
+	parsedURL, err := url.Parse(mbusURL)
+	if err != nil {
+		return "", bosherr.WrapError(err, "Parsing Mbus URL")
+	}
+	var username, password, port string
+	_, port, _ = net.SplitHostPort(parsedURL.Host)
+	userInfo := parsedURL.User
+	if userInfo != nil {
+		username = userInfo.Username()
```

Comment:
I like seeing functions end with their return statement, so I'd say rewrite this last if-else as:

```go
if userInfo != nil {
		username = userInfo.Username()
		password, _ = userInfo.Password()
		return fmt.Sprintf("https://%s:%s@%s:%s", username, password, primaryBackendIpAddress, port), nil
}

return fmt.Sprintf("https://%s:%s", primaryBackendIpAddress, port), nil
```